test/e2e/apps/replica_set.go

```diff
@@ -495,3 +502,48 @@ func testRSLifeCycle(f *framework.Framework) {
 	framework.ExpectEqual(*(rs.Spec.Replicas), rsPatchReplicas, "replicaset should have 3 replicas")
 	framework.ExpectEqual(rs.Spec.Template.Spec.Containers[0].Image, rsPatchImage, "replicaset not using rsPatchImage. Is using %v", rs.Spec.Template.Spec.Containers[0].Image)
 }
+
+// List and DeleteCollection operations
+func listRSDeleteCollection(f *framework.Framework) {
+
+	ns := f.Namespace.Name
+	c := f.ClientSet
+	rsClient := f.ClientSet.AppsV1().ReplicaSets(ns)
+	zero := int64(0)
+	rsName := "test-rs"
+	replicas := int32(3)
+	e2eValue := rand.String(5)
+
+	// Define ReplicaSet Labels
+	rsPodLabels := map[string]string{
+		"name": "sample-pod",
+		"pod":  WebserverImageName,
+		"e2e":  e2eValue,
+	}
+
+	ginkgo.By("Create a ReplicaSet")
+	rs := newRS(rsName, replicas, rsPodLabels, WebserverImageName, WebserverImage, nil)
+	_, err := rsClient.Create(context.TODO(), rs, metav1.CreateOptions{})
+	framework.ExpectNoError(err)
+
+	ginkgo.By("Verify that the required pods have come up")
+	err = e2epod.VerifyPodsRunning(c, ns, "sample-pod", false, replicas)
+	framework.ExpectNoError(err, "Failed to create pods: %s", err)
+	r, err := rsClient.Get(context.TODO(), rsName, metav1.GetOptions{})
+	framework.ExpectNoError(err, "failed to get ReplicaSets")
+	framework.Logf("Replica Status: %+v", r.Status)
+
+	ginkgo.By("Listing all ReplicaSets")
+	rsList, err := c.AppsV1().ReplicaSets("").List(context.TODO(), metav1.ListOptions{LabelSelector: "e2e=" + e2eValue})
+	framework.ExpectNoError(err, "failed to list ReplicaSets")
+	framework.ExpectEqual(len(rsList.Items), 1, "filtered list wasn't found")
+
+	ginkgo.By("DeleteCollection of the ReplicaSets")
+	err = rsClient.DeleteCollection(context.TODO(), metav1.DeleteOptions{GracePeriodSeconds: &zero}, metav1.ListOptions{LabelSelector: "e2e=" + e2eValue})
```

Comment:
@smarterclayton Can I you confirm the issue with using `GracePeriodSeconds: &zero` and what was the suggested fix? TIA

Reply:
Nothing should delete with grace period zero in normal behavior because it masks the state of the node from the apiserver (delete = 0 means the kubelet is doing more work than the apiserver can see) and is abnormal behavior.  Functionally, delete = 1 is exactly the same outcome (1s vs 0s is immaterial) and both kubelet and apiserver do roughly the same amonut of work, AND delete = 1 has none of the problems of detaching the state of apiserver from node, so we hsould always do 1s in testing instead of 0s unless we are specifically testing scenarios where 0s is relevant.